src/Repository/DefaultUnleashRepository.php

```diff
@@ -91,6 +91,8 @@ public function getFeatures(): iterable
                     if ($response->getStatusCode() === 200) {
                         $data = (string) $response->getBody();
                         $this->setLastValidState($data);
+                    } else {
+                        throw new HttpResponseException("Invalid status code: '{$response->getStatusCode()}'");
```

Comment:
I think this is correct. I don't believe PHP is ever going to do the HTTP 304 dance like the other SDKs and anything in the 500 range should be fine to get handled in the catch.

There's two 400 range HTTP status codes I want to clarify:
401 and 429. Do we want to raise error events in the dispatcher when the client is either being denied due to a bad token or rate limited by our infra? I want to say yes, just a sanity check that you've thought about it and agree

Reply:
304 should be handled automatically and shouldn't be any issue.

429 seems like the perfect use-case for this, I'm a little conflicted on the 401, but ultimately I think that the SDK should try to fail as little as possible.

There is a time limit for the stale cache, meaning if the 401 prevails it will fail with an exception eventually, while temporary outages somewhere along the line (like some proxy having an outage) which could cause the 401 will be handled gracefully.

Reply:
> ultimately I think that the SDK should try to fail as little as possible.

Agree with this.

Not critical, but should they body also be supplied in the exception message to help devs debug errors, in case it's available?

Reply:
@eblanshey I don't think it would be useful given the exception is caught immediately so nothing ever propagates further. Or did you mean somewhere else?

Reply:
>429 seems like the perfect use-case for this, I'm a little conflicted on the 401, but ultimately I think that the SDK should try to fail as little as possible.
Definitely don't think this should cause a failure, idly wondering about the value of being able to respond to a 401 with a hook. Don't think it's harmful to have though so yeah, looks good

Reply:
>There is a time limit for the stale cache, meaning if the 401 prevails it will fail with an exception eventually, while temporary outages somewhere along the line (like some proxy having an outage) which could cause the 401 will be handled gracefully.

Not related to this PR so maybe a conversation for a different time and place but is this what we want? My gut feel is the SDK raising an exception almost anywhere is surprising and not really handleable by the caller. Wouldn't mind unpacking this at some point (honestly I'd like a standard set of behaviours for our SDKs on this, which we don't really have right now)